### ISSUE TITLE
There is no way to use CachedCompilerFilter

### ISSUE THREAD

**Issue by kalekseev** [django-compressor/django-compressor#796]
To use caching user has to add mimetype to the COMPRESS_CACHEABLE_PRECOMPILERS setting and related compiler must be inherited from CachedCompilerFilter (btw the second requirement is not described in the docs). The problem is that CachedCompilerFilter accepts mimetype as a first arg and  CompilerFilter accepts content as a first arg so this call fails https://github.com/django-compressor/django-compressor/blob/6b157a0baaf1c6b33863e767bb49f796054e996d/compressor/base.py#L270

**Comment by rubengrill** [comment]
I don't think that the `CachedCompilerFilter` should (and was meant to) be extended by other python precompilers, as the doc of the `CompilerFilter` states that it is intended to execute external commands:
```
A filter subclass that is able to filter content via external commands.
```

However, python precompilers should also not implement the caching on their own again.
I think that they should not even have to add a cache mixin (possibly provided by django-compressor), because the python precompiler would then be responsible to make sure that the `COMPRESS_CACHEABLE_PRECOMPILERS` setting works as documented in django-compressor.

Is there any reason, why only output of external commands are are supported to be cached in django-compressor?

If not, I would suggest to move the cache logic from the `CachedCompilerFilter` to the `Compressor` as instance method, used twice, before invoking the `CompilerFilter` AND before invoking the `precompiler_class`.
